Spring25 - M3 Milestone - Prepare r2.1 (#37)
camaraproject/DeviceSwap: documentation/API_documentation/device-swap-API-Readiness-Checklist.md

# API Readiness Checklist

Checklist for device-swap 0.2.0-rc.1 in r2.1

> | Nr | API release assets  | alpha | release-candidate |  initial<br>public | stable<br> public | Status | Comments |

Reviewer: For Spring25 the template was slightly updated, e.g. the instructions and wiki link, but also the last column and its expected content:
```suggestion
| Nr | API release assets  | alpha | release-candidate |  initial<br>public | stable<br> public | Status | Reference information  |
```

|----|----------------------------------------------|:-----:|:-----------------:|:-------:|:------:|:----:|:----:|
|  1 | API definition                               |   M   |         M         |    M    |    M   |   Y  | [link](code/API_definitions/device-swap.yaml) |

> |  2 | Design guidelines from Commonalities applied |   O   |         M         |    M    |    M   |   Y  |      |
> |  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |   Y  |      |

Reviewer: Seems that the release candidates of Commonalities and ICM are applied:
```suggestion
|  2 | Design guidelines from Commonalities applied |   O   |         M         |    M    |    M   |   Y  |  r2.2    |
|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |   Y  |  r2.2    |
```

|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |   Y  |      |
|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y  | Embed documentation into API spec - [link](/code/API_definitions/device-swap.yaml) |
|  6 | User stories                                 |   O   |         O         |    O    |    M   |   Y  | [check](/documentation/API_documentation/device-swap_check_User_Story.md) [retrieve](/documentation/API_documentation/device-swap_retrieve_User_Story.md) |
|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y  | [check](/code/Test_definitions/device-swap-check.feature) [retrieve](/code/Test_definitions/device-swap-retrieve-date.feature) |

> |  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   N  | pending |

Reviewer: not mandatory for an initial API Version, so you can left the last column blank.

> |  9 | Test result statement                        |   O   |         O         |    O    |    M   |   N  |  Spring25 EXCEPTION: Test results not available (*)    |

Reviewer: That exception is for Spring25 not yet decided within the TSC ... and the test result statement is only needed for public releases stable API, not for the release candidates and also not for  initial API versions like 0.2.0.
Therefore:
```suggestion
|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   N  |           |
```

| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |   Y  |      |
| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y  | [link](/CHANGELOG.md) |
| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   N  |     |

(*) If you encounter issues with the provided test files (.feature), please create an issue in the API Sub-Project to signal these issues so they can be fixed in a patch release.

To fill the checklist:

> - in the line above the table, replace the api-name, api-version and the rx.y by their actual values for the current API version and release.
> - in the Status column, put "Y" (yes) if the release asset is available or fulfilled in the current release, a "N" (no) or a "tbd". Example use of "tbd" is in case an alpha or release-candidate API version does not yet provide all mandatory assets for the release.
> - in the Comments column, provide the link to the asset once available, and any other relevant comments.
> 
> Note: the checklists of a public API version and of its preceding release-candidate API version can be the same.
> 
> The documentation for the content of the checklist is here: [API Readiness Checklist](https://wiki.camaraproject.org/x/HQBFAQ)

Reviewer: ```suggestion
- in the line above the table, replace the api-name, api-version and the rx.y by their actual values for the current API version and release.
- in the Status column, put "Y" (yes) if the release asset is available or fulfilled in the current release, a "N" (no) or a "tbd". Example use of "tbd" is in case an alpha or release-candidate API version does not yet provide all mandatory assets for the release.
- in the Reference information column, provide the relative links (from the API repository home folder) to the release asset once available, the applicable release numbers (not versions) of Commonalities and ICM, and any other relevant links or information.
- For the point 12: The Reference information comment shall reference a note (e.g. "see (1)") under the checklist table to be added that states the certified company(s) as can be found on the following link: [GSMA Open Gateway Portal](https://open-gateway.gsma.com/).

Note: the checklists of a public API version and of its preceding release-candidate API version can be the same.

The documentation for the content of the checklist is here: see API Readiness Checklist section in the [API Release Process](https://lf-camaraproject.atlassian.net/wiki/x/jine).
```
